File: cpp/estructura_de_datos/data_structures/Queue/java/queue/Queue.java
public class Queue {
	
	private Node head;
	private Node tail;

	public void enqueue(int n) {
		Node temp = new Node(n); // Se crea el nodo
		if (isEmpty()) {
			head = temp;
			tail = temp;
		} else {
			Node current = head;
			while(current.getNext() != null) {
				current = current.getNext();
			}
			current.setNext(temp);
			tail = temp;
			tail.setNext(null);
		}
	}

	public int dequeue() {
		if (!isEmpty()) {
			int n = head.getLoad();
			Node temp = head.getNext();
			head = temp;
			return n;
		} else return -1;
	}

	public void clear() {
		Node current = head;
		while(!isEmpty()) dequeue();
		head = null;
		tail = null;
	}

	public int peek() {
		return head.getLoad();
	}

	public boolean isEmpty() {
		if (head == null) return true;
		else return (head.getNext() == null);
	}

	public String toString() {
		String s = "[ ";
		while (isEmpty()) {
			s += dequeue();
			s += ", ";
		}
		s += " ]";
		return s;
	}

}

File: cpp/estructura_de_datos/data_structures/Queue/cpp/queue/with_classes/Queue.cpp
#include "Queue.h"
#include <iostream>

Queue::Queue()
{
	head = tail = 0;
}

bool Queue::isEmpty()
{
	if(head == 0) return true;
	else return false;
}

void Queue::enqueue(int n)
{
	Node nuevoNodo(n);
	if(isEmpty()) {
		head = &nuevoNodo;
		tail = &nuevoNodo;
	} else {
		tail->setNext(&nuevoNodo);
		tail = &nuevoNodo;
	}
}

int Queue::dequeue()
{
	if (isEmpty() == true) return -1;
	else {
		int aux = head->getLoad();
		Node *a = head;
		head = a->getNext();
		return aux;
	}
}

int Queue::peek()
{
	if(isEmpty() == true) return -1;
	else return head->getLoad();
}
